tasks/ci.rake

```diff
@@ -243,13 +256,37 @@ end
 def beaker_suite(type)
   beaker(:init, '--hosts', ENV['HOSTS'], '--options-file', "config/#{String(type)}/options.rb")
   beaker(:provision)
-  beaker(:exec, 'pre-suite', '--preserve-state', '--pre-suite', pre_suites(type))
-  beaker(:exec, 'pre-suite', '--preserve-state')
-  beaker(:exec, ENV['TESTS'])
-  beaker(:exec, 'post-suite')
+  begin
+    beaker(:exec, 'pre-suite', '--preserve-state', '--pre-suite', pre_suites(type))
+    beaker(:exec, 'pre-suite', '--preserve-state')
+    beaker(:exec, ENV['TESTS'])
+    beaker(:exec, 'post-suite')
+  ensure
+    preserve_hosts = ENV['OPTIONS'].include?('--preserve-hosts=always') if ENV['OPTIONS']
+    beaker(:destroy) unless preserve_hosts
+  end
+end
+
+def beaker_suite_retry(type)
+  beaker(:init, '--hosts', ENV['HOSTS'], '--options-file', "config/#{String(type)}/options.rb")
+  beaker(:provision)
+  begin
+    beaker(:exec, 'pre-suite', '--preserve-state', '--pre-suite', pre_suites(type))
+    beaker(:exec, 'pre-suite', '--preserve-state')
+
+    beaker(:exec, ENV['TESTS'], '--save-results', '/tmp/results.json')
```

Comment:
I believe that this should be using something like Tempfile for the interim storage. What do you think? Also, based on my reading of how Tempfile works, if we go this route, I wasn't going to explicitly remove the file and let Tempfile do it's own [cleanup](https://www.rubydoc.info/stdlib/tempfile/Tempfile).

Reply:
Yes, an auto-created tempfile would make sense. If finding the path for that file again later is an issue, perhaps it could be recorded  in beaker's subcommand_options file on creation?

Reply:
Done.